core/Plugin/Manager.php

```diff
@@ -1076,11 +1076,20 @@ private function installPluginIfNecessary(Plugin $plugin)
 
         if ($saveConfig) {
             PiwikConfig::getInstance()->forceSave();
+            $this->clearCache($pluginName);
         }
     }
 
     public function isTrackerPlugin(Plugin $plugin)
     {
+        if (!$this->isPluginInstalled($plugin->getPluginName())) {
+            return false;
+        }
+        
+        if ($plugin->isTrackerPlugin()) {
```

Comment:
Moved it further up as a performance tweak, this way we do not have to check for all dimensions, events, ... when a plugin directly marks itself as a tracker plugin